### DISCUSSION TITLE
Chain conversational_retrieval_chain expects multiple inputs, cannot use 'run'

### DISCUSSION THREAD

**Discussion by JoeLee6286** [FlowiseAI/Flowise#376] [Q&A] [answered]
Chain conversational_retrieval_chain expects multiple inputs, cannot use 'run' 

How can i resolve this issue?

<img width="870" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/125793578/e943dce9-f822-4bf2-b0d1-1a63dd5bcd58">

**Comment by oswaldB** [comment]
Hello, I have exactly the same issue. I am doing something very similar.

Thank you for your response.

**Reply by oswaldB** [reply]
Do we know if it is something possible with langchain?

**Comment by HenryHengZJ** [comment]
you can only use `RetrievalQAChain` when connect to `ChainTool`

**Reply by evmond1** [reply]
Thats what I have and yet I  get the same error

**Reply by ErikEsins** [reply]
I think it is setup i an way that does only allow one Agent or one QA Chain. I wanted to add my Knoledgebase as optional Memory to an Agent aswell but get the same Error. Its what everyone whats to do atm so we might just have to wait until they add a new Tool for it like PineconeVectorBase as Memory or something like that.

**Comment by ErikEsins** [comment, marked as answer]
I had the same Problem. The Error is in the Fact that there is a ConversationalQ&Q-Chain and a normal RetrivalQ&Q-Chain. The Conversational Chain has a LLM baked in i think? The other one can be used as a Tool for Agents.

I used mine so my Agent can use the Pinecone VetorBase of me shouldh he need to load some Information into the buffer memory. As far as i can tell You can not use a Conversational Retrival and a Agent at the same time. Here is a working example: 

Hope this fixes your Problem and have a good one!
![Desktop Screenshot 2023 06 29 - 22 15 59 82](https://github.com/FlowiseAI/Flowise/assets/117544460/12d14be7-7547-4a39-8440-70af61d24111)

I also posted a working Example on the official FlowiseAI Discord you can just download and use it. Just search for "Memory via chain tool for QnA chat" its a Thread you will find it.

**Comment by trepont** [comment]
But how to use return source documents in this flow?
